src/main/com/yetanalytics/lrs/pedestal/interceptor/xapi/statements.cljc

```diff
@@ -236,6 +236,33 @@
                 {:status 400
                  :body {:error {:message "No Statement Data Provided"}}}))))})
 
+(defn scan-attachments
+  "Scan attachment files with a user-provided function."
+  [file-scanner]
+  {:name ::scan-attachments
+   :enter
+   (fn [ctx]
+     (let [attachments (get-in ctx [:xapi :xapi.statements/attachments])]
+       (if-let [attachment-errors
+                (some-> attachments
+                        (->> (keep (fn [{:keys [content]}]
+                                     (try
+                                       (file-scanner content)
+                                       (catch #?(:clj Exception :cljs js/Error) _
+                                         {:message "Scan Error"})))))
+                        not-empty)]
+         (do
+           (attachment/delete-attachments! attachments)
```

Comment:
Is there a reason why we have to delete the attachments on failure? Especially when we don't do the same for documents? (I presume the attachments being viruses is the big reason...)

Reply:
Unlike documents which are handled completely in-memory, for attachments we use tempfiles, so this is a case of best-effort not to leave virusy files around

Reply:
d7032ba2b766d134b8981cf039873723f57343ea